src/main/java/dasniko/keycloak/authenticator/SmsAuthenticator.java

```diff
@@ -80,22 +126,194 @@ public void authenticate(AuthenticationFlowContext context) {
 	public void action(AuthenticationFlowContext context) {
 		// check context requirements are defined
 		codeContextIsValid(context);
+		UserModel user = context.getUser();
+		// check user is enabled
+		boolean isEnabled = userIsEnabled(context, user);
+		if (isEnabled) {
+			String isResend = context.getHttpRequest().getDecodedFormParameters().getFirst(RESEND_PARAM);
+			if (Objects.equals(isResend, "resendCode")) {
+				resendCode(context);
+			} else {
+				// check enteredCode is valid
+				validateEnteredCode(context);
+			}
+		} else {
```

Comment:
Reminder to add the smol comment for our future selves about this `else` block potentially not being reachable due to the way the `userIsEnabled` call above works.